src/domain/entities/Log.ts

```diff
@@ -3,20 +3,52 @@ import { ISODateString } from "$/domain/entities/Ref";
 import { User } from "$/domain/entities/User";
 import { Struct } from "$/domain/entities/generic/Struct";
 import _ from "$/domain/entities/generic/Collection";
+import i18n from "$/utils/i18n";
 
 export type LogsAttrs = {
     date: ISODateString;
-    actionDescription: string;
-    action: "sharing" | "orgunits" | "delete" | "edit" | "create" | "clone" | "unknown";
+    action: LogAction;
     user: Pick<User, "id" | "username" | "name">;
     status: LogStatus;
     type: "dataSets";
-    dataSets: Pick<DataSet, "id" | "shortName">[];
+    dataSets: Pick<DataSet, "id" | "name">[];
 };
 
-export type LogStatus = "success" | "failure";
+export type LogAction =
+    | "sharing"
+    | "orgunits"
+    | "delete"
+    | "edit"
+    | "create"
+    | "clone"
+    | "unknown"
+    | "period_dates";
+
+export type LogStatus = "success" | "failed";
+export type PartialLog = Pick<Log, "action" | "status">;
 
 export class Log extends Struct<LogsAttrs>() {
+    get actionDescription(): string {
+        switch (this.action) {
+            case "sharing":
+                return i18n.t("change sharing settings");
+            case "orgunits":
+                return i18n.t("change organisation units");
+            case "delete":
+                return i18n.t("delete");
+            case "edit":
+                return i18n.t("edit dataset");
+            case "create":
+                return i18n.t("create new dataset");
+            case "clone":
+                return i18n.t("clone dataset");
+            case "period_dates":
+                return i18n.t("change period dates");
+            default:
+                return i18n.t("unknown action");
+        }
```

Comment:
Not wrong, but typically a simple Record<ActionKey, string> should work for this kind of mapping.